CONTRIBUTING.md

```diff
@@ -1,26 +1,202 @@
+
+- Use blank lines to split "chunks" of code that make sense together, this will make easier to read where a block of code starts and where it ends.
+- If a line is getting too long, think about breaking it into intermediate steps.
+- If you have a lot of parameters in a function, move them to a "configuration" object.
+- If you are doing multiple operations over a single piece of data, think if it makes sense to restructure the data or to break the operations in smaller steps.
+
+### Avoid clever tricks
+
+Like English where using _hermetic_ words (pun intended) makes us sound fancier, if our code has clever tricks without any reason or explanation, it makes hard for everyone to understand.
+
+### Comments and self-documenting code
+
+Comments exist and are useful to help understand the code, it is not a hard rule to either _"comment everything"_ or _"not comment anything"_. So add them when it helps to understand the code.
+
+As general _suggestions_:
+
+- The code should be readable by itself, that means a person reading it should be able to follow the code without much problem.
+- When you are writing a piece of code that is complex, does something obscure, or does a very specific thing, write a comment.
+- When adding context to a function or variable will help, add a comment.
+
+Typescript and VS Code provide good tooling for JSDoc comments, here is an example of how to document things:
+
+```typescript
+/**
+ * Fetched the user avatar from [Gravatar](https://gravatar.com/).
+ *
+ * The user id is used to get the email first, as gravatar images are fetched using the email.
+ *
+ * @param userId The user UUID in the database.
+ */
+function fetchUserAvatar(userId: string) {
+	// ...
+}
+
+/**
+ * The default timeout used by all fetch requests.
+ */
+const DEFAULT_TIMEOUT_IN_SECONDS = 5;
+```
+
+## Branches and forks
+
+When you are working in a feature it is better to work on a separate branch, that makes the code contained and easier to test by others. If you have access to directly create branches from the repository, do it, if you don't create a fork and then create a new branch.
+
+We don't enforce a naming convention for branches, but we encourage it to be meaningful names that quickly describe what you are working on. It is okay to use prefixes like `fix/`, `feat/`, or `chore/` to represent a bug fix, a feature, or some housekeeping task, respectively.
+
+## Committing
+
+We don't enforce a committing standard, but below are some guidelines for commit messages:
+
+- Be descriptive, a commit message tells what happened to that code.
+- Commit messages should aim to "fit in a Tweet". That adds conciseness.
+- The first line of the commit is the summary, use it!
+- If more context is needed for the changes, write a summary, leave a blank line, then add a longer message as needed.
+- Commit often so you don't have a lot to commit about.
+- The changed files are part of the commit, so no need to be redundant in the message.
```

Comment:
What about adding a guideline to include a prefix in commit messages, following [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)?

Reply:
Sure, I'll add that